docs/development.md

```diff
@@ -1,78 +1,73 @@
 # Development
 
-## golangci-lint
+## Table Of Contents
 
-A [golangci-lint](https://golangci-lint.run/) [config](../.golangci.yaml) and [Makefile](../Makefile) target are provided to keep the codebase aligned to certain best practices and styles.
+- [Running Locally](#running-locally)
+- [Testing](#testing)
+- [API Changes](#api-changes)
+- [Development Hints](#development-hints)
+- [Controllers](#controllers)
+- [Golang CI Linter](#golang-ci-linter)
+- [Pre-commit Hooks](#pre-commit-hooks)
 
-The target installs golangci-lint if it is not already present. To run use the following command:
-
-```shell
-make lint
-```
-
-The target is also used when running unit tests locally.
-
-## pre-commit
-
-A [pre-commit](https://pre-commit.com/) [config](../.pre-commit-config.yaml) is provided to help developers catch any simple mistakes prior to committing their changes.
-
-To install and use the tool please run the following commands:
+## <a name="running-locally"></a> Running Locally
 
-```shell
-pip install --user pre-commit
-pre-commit install
-git commit -s
-```
-
-## Running locally
-
-The operator can run locally on the developer's machine.
-It will watch the cluster configured in a file pointed to by the `$KUBECONFIG`.
+The operator can run locally on the developer's machine,
+monitoring the cluster specified in a file indicated by the `$KUBECONFIG`
+environment variable:
 ```shell
 make install                    # Install CRDs to the cluster
 make run ENABLE_WEBHOOKS=false  # Start the operator locally
 ```
 
-When running locally, the validating webhooks that check the SSP CR
-are disabled. It is up to the developer to use correct SSP CRs.
+When running locally, the validating webhooks responsible for verifying the SSP Custom Resource (CR)
+are deactivated, placing the responsibility on the developer to ensure the usage of the accurate
+SSP CR.
 
 The CRDs can be removed using:
 ```shell
 make uninstall
 ```
 
 Alternatively developers can also deploy `kubevirtci` based clusters before
-building and deploying the operator. For convenience the following makefile
+building and deploying the operator.
+
+For convenience the following Makefile
 targets are provided that automate this flow either directly through the
 `kubevirtci` project or indirectly using the main `kubevirt` project to
-launch an environment. The latter being useful when testing changes that
-rely on core changes in `kubevirt` itself:
+launch an environment.
 
+The latter being useful when testing changes that
+rely on core changes in `kubevirt` itself:
 ```shell
 make cluster-up   # Deploys the latest released version of KubeVirt
 make cluster-sync # Builds and deploys the SSP operator from source
 make cluster-down # Destroys the environment
 ```
-
 ```shell
 make kubevirt-up   # Deploys KubeVirt from source
 make kubevirt-sync # Builds and deploys the SSP operator from source
 make kubevirt-down # Destroys the environment
 ```
 
-## Testing
+## <a name="testing"></a> Testing
```

Comment:
Do we need these anchors? I thought GitHubs Markdown allowed you to reference titles without them?

https://github.com/fefong/markdown_readme#anchor-links

Reply:
Oh okay cool, I didn't know about it. Thanks.